File: vecinapp/backend/app/schemas/unit.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum

# Enums en Español
class TipoUnidad(str, Enum):
    DEPARTAMENTO = "Departamento"
    CASA = "Casa"
    BODEGA = "Bodega"
    ESTACIONAMIENTO = "Estacionamiento"
    ESPACIO_COMUN = "Espacio Común"

class EstadoUnidad(str, Enum):
    DISPONIBLE = "Disponible"
    ASIGNADO = "Asignado"

class UnitBase(BaseModel):
    name: str  # Ej: "101", "12B"
    floor: int
    type: TipoUnidad = TipoUnidad.DEPARTAMENTO
    status: EstadoUnidad = EstadoUnidad.DISPONIBLE
    
    # Datos Financieros / Técnicos
    alicuota: float = 0.0 # Porcentaje de participación en gastos comunes
    m2: float = 0.0       # Metros cuadrados
    
    description: Optional[str] = None
    community_id: str

class UnitCreate(UnitBase):
    pass

class Unit(UnitBase):
    id: str
    
    class Config:
        from_attributes = True

File: vecinapp/admin_web/lib/models/unit.dart
class Unit {
  final String id;
  final String name;
  final int floor;
  final String type;
  final String status;
  final double alicuota;
  final double m2;
  final String? description;
  final String communityId;

  Unit({
    required this.id,
    required this.name,
    required this.floor,
    required this.type,
    required this.status,
    required this.alicuota,
    required this.m2,
    this.description,
    required this.communityId,
  });

  factory Unit.fromJson(Map<String, dynamic> json) {
    return Unit(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      floor: json['floor'] ?? 0,
      type: json['type'] ?? 'Departamento',
      status: json['status'] ?? 'Disponible',
      alicuota: (json['alicuota'] ?? 0).toDouble(),
      m2: (json['m2'] ?? 0).toDouble(),
      description: json['description'],
      communityId: json['community_id'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'floor': floor,
      'type': type,
      'status': status,
      'alicuota': alicuota,
      'm2': m2,
      'description': description,
      'community_id': communityId,
    };
  }
}
